src/pages/SearchPage/index.jsx

```diff
@@ -0,0 +1,192 @@
+   * @property {Function} handleTermChange - handle term
+   * @property {Function} handleSearch - process the search
+   * @property {Function} handleIncomingSearch - handle the search on the global header
+   * @property {Function} renderList - render searched item
+   *
+   */
+  static propTypes = {
+    search: PropTypes.instanceOf(Object).isRequired,
+    searchArticlesRequest: PropTypes.func.isRequired,
+    history: PropTypes.instanceOf(Object).isRequired,
+  };
+
+  constructor(props) {
+    super(props);
+    const {
+      history: {
+        location: { search },
+      },
+    } = this.props;
+    const query = search.split('=')[1];
+    this.state = {
+      term: 'tag',
+      query,
+      searched: false,
+    };
+  }
+
+  componentDidMount() {
+    const { query } = this.state;
+    if (query) {
+      this.handleIncomingSearch();
+    }
+  }
+
+  submitForm = (term) => {
+    const { query } = this.state;
+    const { searchArticlesRequest: searchArticles } = this.props;
+    searchArticles({ term, query });
+  };
+
+  handleInputChange = ({ target: { value } }) => {
+    this.setState({ query: value });
+  };
+
+  handleTermChange = ({ target: { value } }) => {
+    this.setState(() => ({
+      term: value,
+    }));
+
+    this.submitForm(value);
+  };
+
+  handleSearch = ({ keyCode }) => {
+    const { term } = this.state;
+    if (keyCode === 13) {
+      this.setState(() => ({
+        searched: true,
+      }));
+
+      return this.submitForm(term);
+    }
+    return null;
+  };
+
+  handleIncomingSearch = () => {
+    const { term } = this.state;
+    this.setState(() => ({
+      searched: true,
+    }));
+
+    this.submitForm(term);
+  };
+
+  renderList = () => {
+    const {
+      search: { data }
+    } = this.props;
+
+    return (data)
+      ? data.map((article) => (
+        <Link
+          to={`/articles/${article.slug}`}
+          key={article.title}
+          className="card-display"
+        >
+          <Card
+            img={article.coverImageUrl}
+            tag={article.category}
+            title={article.title}
+            summary={article.summary}
+            author={article.author.name}
+          />
+        </Link>
+      ))
+      : [];
+  };
+
+  render() {
+    const {
+      search: { loading },
+    } = this.props;
+    const { query, searched } = this.state;
+
+    return (
+      <div className="search-container">
+        <div>
+          <Search
+            className="search-container__search"
+            type="text"
+            placeholder="Search for articles"
+            searchkeyUp={this.handleSearch}
+            changeValue={this.handleInputChange}
+            searchValue={query}
+            required
+          />
+        </div>
+        {loading && <Spinner caption="Searching" />}
+        {searched && (
+          <div className="search-container__query-button">
+            <button
+              className="query-button__tag"
```

Comment:
Please read up on BEM
```suggestion
              className="button-link query-button query-button--tag"
```
same feedback to the ones under. These are buttons that look like links so please create a new unit component called `LinkButton` which would be properly styled and can be reused throughout the app.
```javascript
<LinkButton className="query-button query-button--tag" onClick={...} {...otherProps}>whatever...</LinkButton>
```
This would also be added to storybook as a type of button under our button stories